### ISSUE TITLE
Is there a way to use date-fns in addition to date-fns-jalali? (without importing date-fns separately)

### ISSUE THREAD

**Issue by smfaramarzirad** [date-fns-jalali/date-fns-jalali#67]
Is there a way to use date-fns in addition to date-fns-jalali? (without importing date-fns separately)

For example

`
import { formatJ, format } from 'date-fns-jalali';
`
`formatJ `should work in Jalali format and `format `in Gregorian format.

Or a solution like [moment-jalaali](https://www.npmjs.com/package/moment-jalaali).

**Comment by smfaramarzirad** [comment]
I understand import renaming. I meant other than that.
Thanks for introducing date-io.
Am I understanding the purpose of date-io correctly? Does it provide the same methods and properties to work with all date libraries?

> You can rename imports
> 
> ```ts
> import { format } from 'date-fns';
> import { format as formatJalali } from 'date-fns-jalali';
> ```
> 
> The moment-jalaali approach is not an option.
> 
> I prefer the [`date-io`](https://github.com/dmtrKovalenko/date-io/) approach for multi-calendar apps. Define an interface that covers all of your requirements, Then implement adapters with `date-fns` and `date-fns-jalali`

I understand import renaming. I meant other than that.
Thanks for introducing date-io.
Am I understanding the purpose of date-io correctly? Does it provide the same methods and properties to work with all date libraries?

**Comment by smfaramarzirad** [comment]
I understand import renaming. I meant other than that.
Thanks for introducing date-io.
Am I understanding the purpose of date-io correctly? Does it provide the same methods and properties to work with all date libraries?

**Comment by smmoosavi** [comment]
> Am I understanding the purpose of date-io correctly? Does it provide the same methods and properties to work with all date libraries?

Yes, you’re on the right track! date-io serves as an abstraction layer for common JavaScript date management libraries. It allows developers to choose which date library they want to use.

A few years ago, date-io was specifically designed as an abstract layer for mui-date-picker. However, since then, mui-date-picker has been moved to the mui repository, and date-io has evolved into a more general-purpose library.

> Abstraction over common javascript date management libraries

**Comment by smfaramarzirad** [comment]
Great. thanks.

**Comment by smfaramarzirad** [comment]
@smmoosavi
I use typescript. I get an error when compiling. After compiling the JavaScript app works correctly. can you help me?
When I try to create an instance of DateFnsAdapter I get an error!

```
import DateFnsAdapter from '@date-io/date-fns-jalali';
const dateFns = new DateFnsAdapter();

dateFns.formatByString(
 new Date(),
  'yyyy-MM-dd HH:mm:ss.SSSX',
)
```

This expression is not constructable.
  Type 'typeof import("c:/myFolder/node_modules/@date-io/date-fns-jalali/build/index")' has no construct signatures.ts(2351)
import DateFnsAdapter
